pkg/metricprovider/aks/trackers/pricing.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package trackers
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/karpenter/pkg/providers/pricing"
+	"github.com/Azure/karpenter/pkg/providers/pricing/client"
+)
+
+// PricingProvider is an interface that the AKS metric provider uses to sync pricing information.
+//
+// It helps decouple the AKS metric provider from the actual pricing provider implementation; even
+// though at this moment the metric provider leverages the AKS Karpenter pricing client only.
+type PricingProvider interface {
+	OnDemandPrice(instanceType string) (float64, bool)
+	LastUpdated() time.Time
+}
+
+var _ PricingProvider = &AKSKarpenterPricingClient{}
+
+// AKSKarpenterPricingClient is a thin wrapper around the AKS Karpenter pricing client, which
+// implements the PricingProvider interface.
+type AKSKarpenterPricingClient struct {
+	kp *pricing.Provider
+}
+
+// OnDemandPrice returns the on-demand price of an instance type.
+func (k *AKSKarpenterPricingClient) OnDemandPrice(instanceType string) (float64, bool) {
+	return k.kp.OnDemandPrice(instanceType)
+}
+
+// LastUpdated returns the last time the pricing information was updated.
+func (k *AKSKarpenterPricingClient) LastUpdated() time.Time {
+	return k.kp.OnDemandLastUpdated()
+}
+
+func NewAKSKarpenterPricingClient(ctx context.Context, region string) *AKSKarpenterPricingClient {
```

Comment:
is that we need to have one KP client per  region?

Reply:
Yeah, that's how the Karpenter pricing client is built. Though, considering that this provider is for AKS clusters only, and one AKS cluster is bound to one region, this should be alright I assume?